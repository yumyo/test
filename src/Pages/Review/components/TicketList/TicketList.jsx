import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TicketCard from '../TicketCard/TicketCard.jsx';
import './TicketList.scss';

class TicketList extends Component {
    render () {
        const { products, title } = this.props;
        return (
            <div className="ticket-list">
                <h1 key="title">
                    Tickets for {title}
                </h1>
                <div key="list">
                    { products.map((product, index) => (
                        <div key={`listing-${index}`}>
                            <TicketCard
                                product={product}
                            />
                        </div>
                    )) }
                </div>
            </div>
        );
    }
}

TicketList.propTypes = {
    title: PropTypes.string.isRequired,
    products: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            title: PropTypes.string,
            price: PropTypes.decimal,
            quantity: PropTypes.number,
        }),
    ).isRequired,
};

export default TicketList;
