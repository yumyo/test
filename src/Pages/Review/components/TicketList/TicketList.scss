.ticket-list {
  margin: 30px;
  padding: 30px;
  background: #fae3d6;
}
