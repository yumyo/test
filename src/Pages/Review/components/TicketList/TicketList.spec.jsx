import React from 'react';
import { shallow } from 'enzyme';
import TicketList from './TicketList.jsx';

describe('<TicketList />', () => {
    let component;
    const props = {
        title: 'test title',
        products: [
            {
                id: '1',
                title: 'prod 1',
                price: 10.00,
                quantity: 2,
            },
            {
                id: '2',
                title: 'prod 2',
                price: 12.40,
                quantity: 1,
            }
        ],
    }

    beforeEach(() => {
        component = shallow(<TicketList {...props} />);
    });

    it('renders', () => {
        expect(component).toMatchSnapshot();
    });
});
