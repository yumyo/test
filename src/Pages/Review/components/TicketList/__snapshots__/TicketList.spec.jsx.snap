// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TicketList /> renders 1`] = `
<div
  className="ticket-list"
>
  <h1
    key="title"
  >
    Tickets for 
    test title
  </h1>
  <div
    key="list"
  >
    <div
      key="listing-0"
    >
      <Connect(TicketCardComponent)
        product={
          Object {
            "id": "1",
            "price": 10,
            "quantity": 2,
            "title": "prod 1",
          }
        }
      />
    </div>
    <div
      key="listing-1"
    >
      <Connect(TicketCardComponent)
        product={
          Object {
            "id": "2",
            "price": 12.4,
            "quantity": 1,
            "title": "prod 2",
          }
        }
      />
    </div>
  </div>
</div>
`;
